File: app/src/main/java/com/example/vincentwang/rainbowtranslate/store/SearchStoreContract.java
package com.example.vincentwang.rainbowtranslate.store;

import com.example.vincentwang.rainbowtranslate.data.WordMain;

import java.util.Calendar;
import java.util.List;

/**
 * Created by vincentwang on 2017/8/27.
 */

public interface SearchStoreContract {
    interface View{
        void showSearchList(List<WordMain> wordMains);
    }
    interface Presenter{
        void searchbuttonclick(int spinnerposition,Calendar startDay,Calendar endDay);
    }

}

File: app/src/main/java/com/example/vincentwang/rainbowtranslate/store/SearchStoreContract.kt
package com.example.vincentwang.rainbowtranslate.store

import com.example.vincentwang.rainbowtranslate.data.WordMain
import com.example.vincentwang.rainbowtranslate.framework.BaseContract
import java.util.*

/**
 * Created by vincentwang on 2017/8/27.
 */

interface SearchStoreContract {
    interface View:BaseContract.View {
        fun showSearchList(wordMains: List<WordMain>)
    }

    interface Presenter {
        fun searchPeriod(startDay: Calendar?, endDay: Calendar?)
        fun searchToday()
        fun searchAll()
    }
}
